### PR TITLE
Json and Markdown checklist name restructure

### PR DESCRIPTION
**Pull request by rejahrehim**
This PR covers build file restructure 

- [x] This PR handles the issue and requires no additional PRs.
- [x] You have validated the need for this change.

**What did this PR accomplish?**

- Update to a standard structure for build scripts and checklist folder.

### CODE REVIEW

**Comment by rejahrehim** [comment]
> **The following links are broken:** FILE:checklist/README.md [✖] https://raw.githubusercontent.com/OWASP/wstg/master/checklist/checklist.md → Status: 404

Will be created once this PR is merged.

**Review by kingthorin — APPROVED** [review]

**Comment by ThunderSon** [comment]
should we rename the folder to `checklists` instead of `checklist` (added "s")?

Otherwise looks good 👍

**Comment by rejahrehim** [comment]
> should we rename the folder to `checklists` instead of `checklist` (added "s")?
> 
> Otherwise looks good 👍

Done

**Comment by rejahrehim** [comment]
> **The following links are broken:** FILE:checklists/README.md [✖] https://raw.githubusercontent.com/OWASP/wstg/master/checklists/checklist.xlsx → Status: 404 [✖] https://raw.githubusercontent.com/OWASP/wstg/master/checklists/checklist.md → Status: 404

Will be created once this PR is merged.

File: .github/xlsx/scripts/build-checklist.py [hunk 1]
```diff
@@ -7,7 +7,7 @@
 from openpyxl import load_workbook
 from collections import OrderedDict
 
-checklist_template = '.github/xlsx/checklist-template.xlsx'
+checklist_template = '.github/xlsx/assets/checklist-template.xlsx'
```
Comment (kingthorin) [inline .github/xlsx/scripts/build-checklist.py:10]: I suspect there’s a path in this file that needs changing `checklist` ➡️ `checklists`

Comment (kingthorin) [reply .github/xlsx/scripts/build-checklist.py:10]: L95-97

**Comment by ThunderSon** [comment]
QQ, I think the checklists generation build is occurring on any/every change on the repo that's happening. Is the latest action addition we have running on every commit? 🤔

File: .github/xlsx/scripts/build-checklist.py (hunk 1 above)
Comment (rejahrehim) [reply .github/xlsx/scripts/build-checklist.py:10]: Fixed

**Comment by rejahrehim** [comment]
> QQ, I think the checklists generation build is occurring on any/every change on the repo that's happening. Is the latest action addition we have running on every commit? 🤔

Runs on master branch if any changes in `document` folder.

```
on:
  push:
    branches:
      - master
    paths:
      - 'document/**'
 ```

**Review by kingthorin — APPROVED** [review]

**Review by ThunderSon — APPROVED** [review]

**Pull request merged**